frontend/wallet/src/render/components/SideBar.re

```diff
@@ -40,53 +40,56 @@ let make = () => {
     React.useContext(AddressBookProvider.context);
 
   <div className=Styles.sidebar>
-    <WalletQuery>
-      {response =>
-         switch (response.result) {
-         | Loading => React.string("...")
-         | Error(err) => React.string(err##message)
-         | Data(data) =>
-           <WalletList
-             wallets={Array.map(~f=Wallet.ofGraphqlExn, data##ownedWallets)}
-           />
-         }}
-    </WalletQuery>
-    <div className=Styles.footer>
-      <Link onClick={_ => setModalState(_ => Some("My Wallet"))}>
-        {React.string("+ Add wallet")}
-      </Link>
-    </div>
-    <AddWalletMutation>
-      {(mutation, _) =>
-         switch (modalState) {
-         | None => React.null
-         | Some(newWalletName) =>
-           <AddWalletModal
-             walletName=newWalletName
-             setModalState
-             onSubmit={name => {
-               let performMutation =
-                 Task.liftPromise(() =>
-                   mutation(~refetchQueries=[|"getWallets"|], ())
+    // TODO: Remove fetchPolicy="no-cache" after merge of
+    // https://github.com/apollographql/reason-apollo/pull/196
+
+      <WalletQuery fetchPolicy="no-cache" partialRefetch=true>
+        {response =>
+           switch (response.result) {
+           | Loading => React.string("...")
+           | Error(err) => React.string(err##message)
+           | Data(data) =>
+             <WalletList
+               wallets={Array.map(~f=Wallet.ofGraphqlExn, data##ownedWallets)}
+             />
+           }}
+      </WalletQuery>
+      <div className=Styles.footer>
+        <Link onClick={_ => setModalState(_ => Some("My Wallet"))}>
```

Comment:
One of the error states is trying to save wallet name that already exists - we may want to think about incrementing this based on # of wallets or something, so we don't default name them all "My wallet"

Reply:
Yeah that makes sense
One thing to note is that nothing will break technically if you name 2 wallets the same thing -- idk if it's necessarily a priority to disable it

Reply:
I'm just thinking we want to be defensive from a product standpoint if we're using an address book, because we don't want people confusing their wallets and transferring funds from the wrong one / forgetting which ones are staking / compressing